app/code/core/Mage/Core/Model/Translate.php

```diff
@@ -422,10 +422,12 @@ public function translate($args)
             $translated = $this->_getTranslatedString($text, $code);
         }
 
-        //array_unshift($args, $translated);
-        //$result = @call_user_func_array('sprintf', $args);
+        try {
+            $result = !empty($args) ? vsprintf($translated, $args) : false;
+        } catch (ValueError $e) {
+            $result = false;
+        }
 
-        $result = !empty($args) ? @vsprintf($translated, $args) : false;
         if ($result === false) {
             $result = $translated;
         }
```

Comment:
In PHP8, `vsprintf()` no longer no longer returns `false` on failure. Ref PR #978, 

![image](https://user-images.githubusercontent.com/1106470/170902576-32fc4084-4ee9-4dc9-a815-9177391046ac.png)

We can simplify the code to
```suggestion
        try {
            $result = vsprintf($translated, $args);
        } catch (ValueError $e) {
            $result = $translated;
        }
```

Reply:
Don't we loose PHP 7.3 compatibility with this simplification?